packages/core-db/src/app/ethereum/ethereum-block-processor.ts

```diff
@@ -112,11 +112,28 @@ export class EthereumBlockProcessor {
         `Waiting for ${this.confirmsUntilFinal} confirms before disseminating block ${blockNumber}`
       )
       try {
-        const receipt: TransactionReceipt = await provider.waitForTransaction(
-          (block.transactions[0] as any).hash,
-          this.confirmsUntilFinal
-        )
-        if (receipt.blockHash !== block.hash) {
+        let refetchedHash: string
+        if (block.transactions.length > 0) {
+          const receipt: TransactionReceipt = await provider.waitForTransaction(
+            (block.transactions[0] as any).hash,
+            this.confirmsUntilFinal
+          )
+          refetchedHash = receipt.blockHash
+        } else {
+          while (
+            (await provider.getBlockNumber()) <
+            blockNumber + this.confirmsUntilFinal
+          ) {
+            log.debug(
+              `Waiting for empty block ${blockNumber} to be final. Sleeping...`
+            )
+            await sleep(15_000)
+          }
```

Comment:
do we need both cases or can we just keep this one?

Reply:
No, don't need to. Was thinking maaaaaybe ethers does something better than just polling, so that's better than this? Happy to consolidate if you think though. Eventually we'll go with the "this block finalizes X blocks ago" approach vs the "I got a block, now wait a bunch of blocks for _this_ block to be final" approach.

Reply:
Going to ship for now but happy to change it post-merge if you'd like!